File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left, right = 0 , len(heights) - 1
        maximum = 0
        current = 0

        while left < right : 
            if heights[left] > heights[right]:
                current = heights[right] * (right - left)
            else:
                current = heights[left] * (right - left)
            if current > maximum:
                maximum = current

            if heights[left] > heights[right]:
                right = right - 1
            elif heights[left] <= heights[right]:
                left = left + 1
            
        return maximum


        

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.java
class Solution {
    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        if (root == null) return false;
        if (isSameTree(root, subRoot)) return true;
        return isSubtree(root.left, subRoot) || isSubtree(root.right, subRoot);
    }

    public boolean isSameTree(TreeNode r, TreeNode s) {
        if (r == null && s == null) return true;
        if (r == null || s == null) return false;
        if (r.val != s.val) return false;
        return isSameTree(r.left, s.left) && isSameTree(r.right, s.right);
    }
}